modules/oke/locals.tf

```
@@ -31,6 +31,16 @@ locals {
 
   # kubernetes string version length
   k8s_version_length = length(var.cluster_kubernetes_version)
-  k8s_version_only = substr(var.cluster_kubernetes_version,1,local.k8s_version_length)
-
+  k8s_version_only   = substr(var.cluster_kubernetes_version, 1, local.k8s_version_length)
+
+  # build a list  of nodepools to autoscale in the format expected by cluster autoscaler:
+  # - --nodes:min:max:nodepool_id
+  autoscaling_nodepools = [
+    for nodepool_name, nodepool_parameters in var.node_pools : { 
+      "name" = "${nodepool_name}", 
+      "node_pool_size" = "${nodepool_parameters.node_pool_size}", 
+      "max_node_pool_size" = "${nodepool_parameters.max_node_pool_size}", 
+      "id" = "${lookup(lookup(oci_containerengine_node_pool.nodepools, nodepool_name), "id")}"       
+    } if nodepool_parameters.autoscale == true
```

>│ Error: Unsupported attribute
│ 
│   on modules/oke/locals.tf line 44, in locals:
│   44:     } if nodepool_parameters.autoscale == true
│ 
│ This object does not have an attribute named "autoscale".

```suggestion
    } if lookup(nodepool_parameters, "autoscale", false)
```

---

The PR is out-of-date. I was waiting on @hyder to finish testing the branch after we rebased it from `main` yesterday.